flavors/benchmark/builder/builder.go

```diff
@@ -0,0 +1,109 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package builder
+
+import (
+	"context"
+	"time"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/elastic-agent-libs/logp"
+
+	"github.com/elastic/cloudbeat/config"
+	"github.com/elastic/cloudbeat/dataprovider"
+	"github.com/elastic/cloudbeat/dataprovider/providers/common"
+	"github.com/elastic/cloudbeat/evaluator"
+	"github.com/elastic/cloudbeat/resources/fetching"
+	"github.com/elastic/cloudbeat/resources/fetching/manager"
+	"github.com/elastic/cloudbeat/resources/fetching/registry"
+	"github.com/elastic/cloudbeat/transformer"
+	"github.com/elastic/cloudbeat/uniqueness"
+	"github.com/elastic/cloudbeat/version"
+)
+
+const (
+	defaultManagerTimeout = 10 * time.Minute
+)
+
+type Benchmark interface {
+	Run(ctx context.Context) (<-chan []beat.Event, error)
+	Stop()
+}
+
+type Builder struct {
+	managerTimeout   time.Duration
+	idp              dataprovider.IdProvider
+	bdp              dataprovider.CommonDataProvider
+	k8sLeaderElector uniqueness.Manager
+}
+
+func New(options ...Option) *Builder {
+	b := &Builder{
+		managerTimeout:   defaultManagerTimeout,
+		idp:              &idProvider{},
+		bdp:              &dataProvider{},
```

Comment:
It's a bit confusing that those struct implementations reside under builder.
consider shifting the "base" implementation to be under the relevant utility.

Reply:
I agree that it may not should be under the builder package however I think those implementations should remain private. Maybe changing the whole package name?